src/main/java/com/google/sps/servlets/ListOrganizationsServlet.java

```diff
@@ -96,9 +73,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(gson.toJson(requestedOrganizations));
   }
 
-  public Query ConstructQueryForUserInfo(String userID) {
-    Query query = new Query("Distributor").setFilter(new FilterPredicate("moderatorList",
-                    FilterOperator.EQUAL, userID)).addSort("creationTimeStamp", SortDirection.DESCENDING);
+  /* Given a user id, this function checks if that user has a maintainer role in the datastore */
+  public boolean userIsMaintainer(String userId) {
+    //TODO(): Check if given user ID has maintainer role in the datastore
+    return true;
+  }
+
+  /* This function constructs a query based on the request parameters & user's role */
+  public Query getQueryFromParams(HttpServletRequest request, UserService userService) {
+    Query query = new Query("Distributor").addSort("creationTimeStamp", SortDirection.DESCENDING);;
+
+    /* displayUserOrgsParameter is true when user only wants to see orgs they moderate*/
+    String displayUserOrgsParameter = request.getParameter("displayUserOrgs");
+    boolean displayUserOrgs = false;
+    if ((displayUserOrgsParameter != null) && (displayUserOrgsParameter.equals("true"))) {
+      /* If a parameter was sent & is set to 'true', then the displayUserOrgs boolean changes to true */
+      displayUserOrgs = true;
+    }
+
+    boolean isUserLoggedIn = userService.isUserLoggedIn();
+    String userId = userService.getCurrentUser().getUserId();
+    boolean userIsMaintainer = userIsMaintainer(userId);
+
+    if (isUserLoggedIn && displayUserOrgs) {
+      /* If the user is logged in and wants to just see their orgs, get their user ID & index with it*/
+      query.setFilter(new FilterPredicate("moderatorList", FilterOperator.EQUAL, userId));
```

Comment:
Add a TODO: 

can you create a new function in the User class that returns the filter query? (maybe not needed?)

Reply:
I set up a prototype for this TODO in the User class, and then wrote a detailed Github issue about the how/why/what for the filtering backend that addresses what needs to be done with these TODO's